### PR TITLE
[Merged by Bors] - chore: remove `style-exceptions.txt`

### PR DESCRIPTION
**Pull request by grunweg**
This file has been recording exceptions to the text-based style linters in `lint-style.py` and `Linter/TextBased`. The length of this file is meant to reach zero - and now, it indeed has. :tada:

Some of the linter errors were easy to fix (and just required sustained effort), such as some misformatted author lines or missing module documentation. The main remaining entries were exceptions for the "long file linter", which has been rewritten as a syntax linter (in #15610 and #16299). Exceptions are now tracked as `set_option linter.style.longFile <number>` instead, obviating the need for this file.

This implies the need for a file with transient style exceptions is gone. All of the remaining text-based linters should not be introduced in new code. For *permanent* exceptions, the file `nolints-style.txt` continues to exist; that file must (intentionally) be updated by hand.

[zulip discussion](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/RFC.3A.20remove.20.60scripts.2Fstyle-exceptions.2Etxt.60)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: scripts/update_nolints_CI.sh [hunk 1]
```diff
@@ -1,7 +1,6 @@
 #!/usr/bin/env bash
 
-# Check if there are changes to `nolints.json` or `style-exceptions.txt`,
-# and file a PR updating these files otherwise.
+# Check if there are changes to `nolints.json` and file a PR updating these files otherwise.
```
[the remark this answers is not in the document]
Comment (bryangingechen) [reply scripts/update_nolints_CI.sh:3]: This was marked as resolved, but I don't see it implemented - did you disagree with the wording change?

**Comment by bryangingechen** [comment]
How confident are we that we won't need to add new transient exceptions for new linters in the future? Or is the idea that they will also be tracked with `set_option` in the files themselves?

The code otherwise looks like it does what it claims to (but admittedly this is my first time looking at the lean4 style linter code).

File: scripts/update_nolints_CI.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (grunweg) [reply scripts/update_nolints_CI.sh]: Sorry, I meant to incorporate it. I think I force-pushed it away by accident... re-added now!

**Comment by grunweg** [comment]
> How confident are we that we won't need to add new transient exceptions for new linters in the future? Or is the idea that they will also be tracked with `set_option` in the files themselves?
> 
> The code otherwise looks like it does what it claims to (but admittedly this is my first time looking at the lean4 style linter code).

That's the right question to ask. Personally, I am willing to make a bet that
- the linter can be fixed beforehand (indeed, that's the normal procedure for syntax linters), or
- these linters won't appear, or
- we can always resurrect the exceptions file. (The logic for parsing the file is still there, to support `nolints-style.txt`. Just the automated infrastructure is gone, and the need for `lint-style.py` to look at the exceptions file.)

**Comment by joneugster** [comment]
I agree with @grunweg, currently it is quite common that a new linter lives in a PR until most of the things it marks have been addressed, which is a good habit. And the PR description now provides good reasoning about future exemptions

I think it's safe to remove the file and drop one more technical depth.

maintainer merge

**Comment by urkud** [comment]
Thanks! :tada:
bors merge

**Pull request closed without merging**